### PR TITLE
UCM/TEST: Resore nodelete linker flag for libucm.so

### LINKED ISSUE #3343: Regression when start running OSU - segfault in ibv_open_device
Before OSU start running, I get this crash error:
```
[r1-node17:55335:0:55335] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace ====
    0  /home/amdtest/yuq/install/ucx/lib/libucs.so.0(+0x1db6c) [0x7f92dcfc6b6c]
    1  /home/amdtest/yuq/install/ucx/lib/libucs.so.0(+0x1ddd4) [0x7f92dcfc6dd4]
    2  /lib64/libibverbs.so.1(+0x9dac) [0x7f92df1eedac]
    3  /lib64/libibverbs.so.1(ibv_open_device+0x3cc) [0x7f92df1ef76c]
    4  /home/amdtest/yuq/install/ucx/lib/ucx/libuct_ib.so.0(+0x18191) [0x7f92dcb4d191]
    5  /home/amdtest/yuq/install/ucx/lib/ucx/libuct_ib.so.0(uct_ib_md_open+0xd1) [0x7f92dcb4ee21]
    6  /home/amdtest/yuq/install/ucx/lib/libuct.so.0(uct_md_open+0x69) [0x7f92ddca1a19]
    7  /home/amdtest/yuq/install/ucx/lib/libucp.so.0(ucp_init_version+0x88a) [0x7f92ddec114a]
    8  /home/amdtest/yuq/install/ompi/lib/openmpi/mca_pml_ucx.so(mca_pml_ucx_open+0xc1) [0x7f92de30ba01]
    9  /home/amdtest/yuq/install/ompi/lib/libopen-pal.so.40(mca_base_framework_components_open+0x16e) [0x7f9707e722de]
   10  /home/amdtest/yuq/install/ompi/lib/libmpi.so.40(+0xd6807) [0x7f97084bf807]
   11  /home/amdtest/yuq/install/ompi/lib/libopen-pal.so.40(mca_base_framework_open+0x81) [0x7f9707e7d961]
   12  /home/amdtest/yuq/install/ompi/lib/libmpi.so.40(ompi_mpi_init+0x68f) [0x7f970843821f]
   13  /home/amdtest/yuq/install/ompi/lib/libmpi.so.40(MPI_Init+0x5e) [0x7f9708467e1e]
   14  mpi/pt2pt/osu_bw() [0x4011a9]
   15  /lib64/libc.so.6(__libc_start_main+0xf5) [0x7f9705ea1c05]
   16  mpi/pt2pt/osu_bw() [0x40168f]
```

Bisect to be the regression of
5e6c73eafb1e6559eba2a32
UCS/UCM: Avoid linking UCM loadable modules with "nodelete" flag

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #3343

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9208/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9217/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9221/ for details (Mellanox internal link).

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -1057,6 +1066,7 @@ run_tests() {
 	do_distributed_task 1 4 run_ucp_client_server
 	do_distributed_task 3 4 test_profiling
 	do_distributed_task 3 4 test_dlopen
+	do_distributed_task 3 4 test_dlopen
```
Comment (hoopoepg) [inline contrib/test_jenkins.sh]: test_ucs_load?

Comment (yosefe) [reply contrib/test_jenkins.sh]: yep. merge issue..

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9228/ for details (Mellanox internal link).

File: test/apps/test_dlopen.c [hunk 2]
```diff
@@ -4,26 +4,92 @@
...
+    handle = dlopen(filename, RTLD_NOW);//TODO no GLOBAL
     if (handle == NULL) {
-        fprintf(stderr, "failed to open %s: %m\n", filename);
+        fprintf(stderr, "failed to open %s: %s\n", filename, dlerror());
         return -1;
     }
 
-    printf("done\n");
+    /* init ucm */
+    {
```
Comment (hoopoepg) [inline test/apps/test_dlopen.c]: minor: why {} used here?

File: test/apps/test_dlopen.c [hunk 3]
```diff
@@ -4,26 +4,92 @@
...
+        if (status != UCS_OK) {
+            printf("ucm_set_event_handler() failed\n");
+            return -1;
+        }
+    }
+
+    /* unload ucp */
     dlclose(handle);
+
+    /* relase the memory - could break if UCM is unloaded */
```
Comment (hoopoepg) [inline test/apps/test_dlopen.c]: minor: relase -> release

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9246/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9259/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9270/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9274/ for details (Mellanox internal link).

**Pull request merged**